framework/src/vectorpostprocessors/VectorPostprocessorData.C

```diff
@@ -29,38 +29,57 @@ VectorPostprocessorData::hasVectorPostprocessor(const std::string & name)
 VectorPostprocessorValue &
 VectorPostprocessorData::getVectorPostprocessorValue(const VectorPostprocessorName & vpp_name, const std::string & vector_name)
 {
-  VectorPostprocessorValue * & pp_val = _values[vpp_name][vector_name];
+  _requested_items.emplace(vpp_name + "::" + vector_name);
 
-  if (pp_val == NULL)
-    pp_val = &declareRestartableDataWithObjectName<VectorPostprocessorValue>(vpp_name + "_" + vector_name, "values");
-
-  return *pp_val;
+  return getVectorPostprocessorHelper(vpp_name, vector_name, true);
 }
 
 VectorPostprocessorValue &
 VectorPostprocessorData::getVectorPostprocessorValueOld(const VectorPostprocessorName & vpp_name, const std::string & vector_name)
 {
-  VectorPostprocessorValue * & pp_val = _values_old[vpp_name][vector_name];
-
-  if (pp_val == NULL)
-    pp_val = &declareRestartableDataWithObjectName<VectorPostprocessorValue>(vpp_name + "_" + vector_name, "values_old");
+  _requested_items.emplace(vpp_name + "::" + vector_name);
```

Comment:
oh wait... This _might_ not be bad depending on how it's used. We actually need three different array, or at least one additional flag: these two for creating the sanity checking (making sure that supplied and requested are on parity), but we also need a flag or something to indicate whether or not a dependency should be created for a given pair. Since this implementation isn't finished, I'll leave it for now.